## –ü–æ–ª—É–∑—ë—Ä–æ–≤–∞ –ê–Ω–∞—Å—Ç–∞—Å–∏—è 
## –ú–∞–≥–∏—Å—Ç—Ä–∞—Ç—É—Ä–∞ –¢—é–º–ì–£
## –ü—Ä–∞–∫—Ç–∏—á–µ—Å–∫–æ–µ –∑–∞–¥–∞–Ω–∏–µ 3
# <div style="text-align:center;"><font color="blue">QIIME 2 </font></div>





```python
!conda --version
```

    conda 23.5.0



```python
!ls data_met/sequences
```

    Abacumov-B-13_S13_L001_R1_001.fastq.gz	Abacumov-B-39_S39_L001_R1_001.fastq.gz
    Abacumov-B-13_S13_L001_R2_001.fastq.gz	Abacumov-B-39_S39_L001_R2_001.fastq.gz
    Abacumov-B-14_S14_L001_R1_001.fastq.gz	Abacumov-B-3_S3_L001_R1_001.fastq.gz
    Abacumov-B-14_S14_L001_R2_001.fastq.gz	Abacumov-B-3_S3_L001_R2_001.fastq.gz
    Abacumov-B-15_S15_L001_R1_001.fastq.gz	Abacumov-B-40_S40_L001_R1_001.fastq.gz
    Abacumov-B-15_S15_L001_R2_001.fastq.gz	Abacumov-B-40_S40_L001_R2_001.fastq.gz
    Abacumov-B-16_S16_L001_R1_001.fastq.gz	Abacumov-B-49_S49_L001_R1_001.fastq.gz
    Abacumov-B-16_S16_L001_R2_001.fastq.gz	Abacumov-B-49_S49_L001_R2_001.fastq.gz
    Abacumov-B-1_S1_L001_R1_001.fastq.gz	Abacumov-B-4_S4_L001_R1_001.fastq.gz
    Abacumov-B-1_S1_L001_R2_001.fastq.gz	Abacumov-B-4_S4_L001_R2_001.fastq.gz
    Abacumov-B-25_S25_L001_R1_001.fastq.gz	Abacumov-B-50_S50_L001_R1_001.fastq.gz
    Abacumov-B-25_S25_L001_R2_001.fastq.gz	Abacumov-B-50_S50_L001_R2_001.fastq.gz
    Abacumov-B-26_S26_L001_R1_001.fastq.gz	Abacumov-B-51_S51_L001_R1_001.fastq.gz
    Abacumov-B-26_S26_L001_R2_001.fastq.gz	Abacumov-B-51_S51_L001_R2_001.fastq.gz
    Abacumov-B-27_S27_L001_R1_001.fastq.gz	Abacumov-B-52_S52_L001_R1_001.fastq.gz
    Abacumov-B-27_S27_L001_R2_001.fastq.gz	Abacumov-B-52_S52_L001_R2_001.fastq.gz
    Abacumov-B-28_S28_L001_R1_001.fastq.gz	Abacumov-B-61_S61_L001_R1_001.fastq.gz
    Abacumov-B-28_S28_L001_R2_001.fastq.gz	Abacumov-B-61_S61_L001_R2_001.fastq.gz
    Abacumov-B-2_S2_L001_R1_001.fastq.gz	Abacumov-B-62_S62_L001_R1_001.fastq.gz
    Abacumov-B-2_S2_L001_R2_001.fastq.gz	Abacumov-B-62_S62_L001_R2_001.fastq.gz
    Abacumov-B-37_S37_L001_R1_001.fastq.gz	Abacumov-B-63_S63_L001_R1_001.fastq.gz
    Abacumov-B-37_S37_L001_R2_001.fastq.gz	Abacumov-B-63_S63_L001_R2_001.fastq.gz
    Abacumov-B-38_S38_L001_R1_001.fastq.gz	Abacumov-B-64_S64_L001_R1_001.fastq.gz
    Abacumov-B-38_S38_L001_R2_001.fastq.gz	Abacumov-B-64_S64_L001_R2_001.fastq.gz



```python
!qiime tools import \
  --type 'SampleData[PairedEndSequencesWithQuality]' \
  --input-path data_met/sequences \
  --input-format CasavaOneEightSingleLanePerSampleDirFmt \
  --output-path demux-paired-end.qza
```

    [32mImported data_met/sequences as CasavaOneEightSingleLanePerSampleDirFmt to demux-paired-end.qza[0m
    [0m

## Deblur


```python
#—Ñ–∏–ª—å—Ç—Ä–∞—Ü–∏—è –ø–æ—Å–ª–µ–¥–æ–≤–∞—Ç–µ–ª—å–Ω–æ—Å—Ç–µ–π –Ω–∞ –æ—Å–Ω–æ–≤–µ –æ—Ü–µ–Ω–∫–∏ –∫–∞—á–µ—Å—Ç–≤–∞ 
!qiime quality-filter q-score \
 --i-demux demux-paired-end.qza \
 --o-filtered-sequences demux-filtered.qza \
 --o-filter-stats demux-filter-stats.qza
```

    [32mSaved SampleData[SequencesWithQuality] to: demux-filtered.qza[0m
    [32mSaved QualityFilterStats to: demux-filter-stats.qza[0m
    [0m


```python
#—Å –ø–æ–º–æ—â—å—é deblur –æ–±—ä–µ–¥–∏–Ω—è—é—Ç—Å—è –ø–æ—Ö–æ–∂–∏–µ –ø–æ—Å–ª–µ–¥–æ–≤–∞—Ç–µ–ª—å–Ω–æ—Å—Ç–∏ –∏ —É–¥–∞–ª—è—é—Ç—Å—è –æ—à–∏–±–∫–∏ —Å–µ–∫–≤–µ–Ω–∏—Ä–æ–≤–∞–Ω–∏—è
!qiime deblur denoise-16S \
  --i-demultiplexed-seqs demux-filtered.qza \
  --p-trim-length 120 \
  --o-representative-sequences rep-seqs-deblur.qza \
  --o-table table-deblur.qza \
  --p-sample-stats \
  --o-stats deblur-stats.qza
```

    [32mSaved FeatureTable[Frequency] to: table-deblur.qza[0m
    [32mSaved FeatureData[Sequence] to: rep-seqs-deblur.qza[0m
    [32mSaved DeblurStats to: deblur-stats.qza[0m
    [0m


```python
# —Ç–∞–±—É–ª–∏—Ä–æ–≤–∞–Ω–∏–µ —Å—Ç–∞—Ç–∏—Å—Ç–∏–∫–∏ —Ñ–∏–ª—å—Ç—Ä–∞—Ü–∏–∏ –¥–ª—è –≤–∏–∑—É–∞–ª–∏–∑–∞—Ü–∏–∏. 
# –†–µ–∑—É–ª—å—Ç–∞—Ç—ã —Å–æ—Ö—Ä–∞–Ω—è—é—Ç—Å—è –≤ —Ñ–∞–π–ª–µ demux-filter-stats.qzv.
!qiime metadata tabulate \
  --m-input-file demux-filter-stats.qza \
  --o-visualization demux-filter-stats.qzv
!qiime deblur visualize-stats \
  --i-deblur-stats deblur-stats.qza \
  --o-visualization deblur-stats.qzv
```

    [32mSaved Visualization to: demux-filter-stats.qzv[0m
    [0m[32mSaved Visualization to: deblur-stats.qzv[0m
    [0m

–û—Ü–µ–Ω–∏—Ç—å —ç—Ñ—Ñ–µ–∫—Ç–∏–≤–Ω–æ—Å—Ç—å —Ñ–∏–ª—å—Ç—Ä–∞—Ü–∏–∏ –∏ –∫–∞—á–µ—Å—Ç–≤–æ –¥–∞–Ω–Ω—ã—Ö –º–æ–∂–Ω–æ —Å –ø–æ–º–æ—â—å. –≤–∏–∑—É–∞–ª–∏–∑–∞—Ü–∏–∏ –ø–æ–ª—É—á–µ–Ω–Ω–æ–π —Ç–∞–±–ª–∏—Ü—ã:


![deblur-stats.qzv](Screenshot_20230618_145223.png)


## Add metadata


```python
!sed 's/,/\t/g' map.csv > map.tsv #–ø—Ä–µ–æ–±—Ä–∞–∑–æ–≤–∞–Ω–∏–µ —Ñ–∞–π–ª–∞ –º–µ—Ç–∞–¥–∞–Ω–Ω—ã—Ö map.csv –≤ —Ñ–æ—Ä–º–∞—Ç TSV 
```


```python
!sed -i '1s/Filename/#SampleID/' map.tsv #–∑–∞–º–µ–Ω–∞"Filename" –Ω–∞ "SampleID"
```


```python
# –°–æ–∑–¥–∞–Ω–∏–µ —Å–≤–æ–¥–Ω–æ–π –∏–Ω—Ñ–æ—Ä–º–∞—Ü–∏–∏ –æ —Ç–∞–±–ª–∏—Ü–µ –ø—Ä–∏–∑–Ω–∞–∫–æ–≤
!qiime feature-table summarize \
  --i-table table-deblur.qza \
  --o-visualization table.qzv \
  --m-sample-metadata-file map.tsv

# –°–æ–∑–¥–∞–Ω–∏–µ —Ç–∞–±–ª–∏—Ü—ã —Å –∏–Ω—Ñ–æ—Ä–º–∞—Ü–∏–µ–π –æ –ø–æ—Å–ª–µ–¥–æ–≤–∞—Ç–µ–ª—å–Ω–æ—Å—Ç—è—Ö (–≤–∫–ª—é—á–∞—è –∏–¥–µ–Ω—Ç–∏—Ñ–∏–∫–∞—Ç–æ—Ä—ã –∏ –∫–æ–ª–∏—á–µ—Å—Ç–≤–æ –≤—Ö–æ–∂–¥–µ–Ω–∏–π)
!qiime feature-table tabulate-seqs \
  --i-data rep-seqs-deblur.qza \
  --o-visualization rep-seqs.qzv

```

    [32mSaved Visualization to: table.qzv[0m
    [0m[32mSaved Visualization to: rep-seqs.qzv[0m
    [0m

![sample-frequencies](sample-frequenc.png)
![feature-frequencies](feature-frequenc.png)
![rep-seqs](rep-seq.png)



## Add tree


```python
# QIIME –≤—ã—Ä–∞–≤–Ω–∏–≤–∞–µ—Ç –ø–æ—Å–ª–µ–¥–æ–≤–∞—Ç–µ–ª—å–Ω–æ—Å—Ç–∏ –∏ —Å—Ç—Ä–æ–∏—Ç —Ñ–∏–ª–æ–≥–µ–Ω–µ—Ç–∏—á–µ—Å–∫–æ–µ –¥–µ—Ä–µ–≤–æ: 
!qiime phylogeny align-to-tree-mafft-fasttree \
  --i-sequences rep-seqs-deblur.qza \
  --o-alignment aligned-rep-seqs.qza \
  --o-masked-alignment masked-aligned-rep-seqs.qza \
  --o-tree unrooted-tree.qza \
  --o-rooted-tree rooted-tree.qza
```

    [32mSaved FeatureData[AlignedSequence] to: aligned-rep-seqs.qza[0m
    [32mSaved FeatureData[AlignedSequence] to: masked-aligned-rep-seqs.qza[0m
    [32mSaved Phylogeny[Unrooted] to: unrooted-tree.qza[0m
    [32mSaved Phylogeny[Rooted] to: rooted-tree.qza[0m
    [0m

# Diversity Analysis

–ê–Ω–∞–ª–∏–∑ —Ä–∞–∑–Ω–æ–æ–±—Ä–∞–∑–∏—è –º–∏–∫—Ä–æ–±–∏–æ–º–∞ –Ω–∞ —É—Ä–æ–≤–Ω–µ –∞–ª—å—Ñ–∞- –∏ –±–µ—Ç–∞-—Ä–∞–∑–Ω–æ–æ–±—Ä–∞–∑–∏—è 



```python
#QIIME —Ä–∞—Å—Å—á–∏—Ç—ã–≤–∞–µ—Ç —Ä–∞–∑–ª–∏—á–Ω—ã–µ –º–µ—Ç—Ä–∏–∫–∏ –∞–ª—å—Ñ–∞- –∏ –±–µ—Ç–∞-—Ä–∞–∑–Ω–æ–æ–±—Ä–∞–∑–∏—è:
!qiime diversity core-metrics-phylogenetic \
  --i-phylogeny rooted-tree.qza \
  --i-table table-deblur.qza \
  --p-sampling-depth 8000 \
  --m-metadata-file map.tsv \
  --output-dir core-metrics-results
```

    [32mSaved FeatureTable[Frequency] to: core-metrics-results/rarefied_table.qza[0m
    [32mSaved SampleData[AlphaDiversity] to: core-metrics-results/faith_pd_vector.qza[0m
    [32mSaved SampleData[AlphaDiversity] to: core-metrics-results/observed_features_vector.qza[0m
    [32mSaved SampleData[AlphaDiversity] to: core-metrics-results/shannon_vector.qza[0m
    [32mSaved SampleData[AlphaDiversity] to: core-metrics-results/evenness_vector.qza[0m
    [32mSaved DistanceMatrix to: core-metrics-results/unweighted_unifrac_distance_matrix.qza[0m
    [32mSaved DistanceMatrix to: core-metrics-results/weighted_unifrac_distance_matrix.qza[0m
    [32mSaved DistanceMatrix to: core-metrics-results/jaccard_distance_matrix.qza[0m
    [32mSaved DistanceMatrix to: core-metrics-results/bray_curtis_distance_matrix.qza[0m
    [32mSaved PCoAResults to: core-metrics-results/unweighted_unifrac_pcoa_results.qza[0m
    [32mSaved PCoAResults to: core-metrics-results/weighted_unifrac_pcoa_results.qza[0m
    [32mSaved PCoAResults to: core-metrics-results/jaccard_pcoa_results.qza[0m
    [32mSaved PCoAResults to: core-metrics-results/bray_curtis_pcoa_results.qza[0m
    [32mSaved Visualization to: core-metrics-results/unweighted_unifrac_emperor.qzv[0m
    [32mSaved Visualization to: core-metrics-results/weighted_unifrac_emperor.qzv[0m
    [32mSaved Visualization to: core-metrics-results/jaccard_emperor.qzv[0m
    [32mSaved Visualization to: core-metrics-results/bray_curtis_emperor.qzv[0m
    [0m

Visualise alpha-diversity


```python
!qiime diversity alpha-rarefaction \
  --i-table table-deblur.qza \
  --i-phylogeny rooted-tree.qza \
  --p-max-depth 20000 \
  --m-metadata-file map.tsv \
  --o-visualization alpha-rarefaction.qzv
```

    [32mSaved Visualization to: alpha-rarefaction.qzv[0m
    [0m

![alpha-rarefaction](alpha-rarefaction.png)
![bray-curtis](bray.png)


```python
#—Å—Ç–∞—Ç–∏—Å—Ç–∏—á–µ—Å–∫–∞—è –æ—Ü–µ–Ω–∫–∞ –∑–Ω–∞—á–∏–º–æ—Å—Ç–∏ –∞–ª—å—Ñ–∞-—Ä–∞–∑–Ω–æ–æ–±—Ä–∞–∑–∏—è –º–µ–∂–¥—É –≥—Ä—É–ø–ø–∞–º–∏ (–Ω–∞ –æ—Å–Ω–æ–≤–µ –≤–µ–∫—Ç–æ—Ä–æ–≤ –∞–ª—å—Ñ–∞-—Ä–∞–∑–Ω–æ–æ–±—Ä–∞–∑–∏—è):
!qiime diversity alpha-group-significance \
  --i-alpha-diversity core-metrics-results/observed_features_vector.qza \
  --m-metadata-file map.tsv \
  --o-visualization core-metrics-results/observed_features-significance.qzv

!qiime diversity alpha-group-significance \
  --i-alpha-diversity core-metrics-results/evenness_vector.qza \
  --m-metadata-file map.tsv \
  --o-visualization core-metrics-results/evenness-group-significance.qzv

```

    [32mSaved Visualization to: core-metrics-results/observed_features-significance.qzv[0m
    [0m[32mSaved Visualization to: core-metrics-results/evenness-group-significance.qzv[0m
    [0m

![alpha-diversity-boxplots](alpha-diversity-boxplots.png)
![kruskal-wallis](kruskal-wallis.png)

–¢–µ—Å—Ç Kruskal-Wallis:
–ù—É–ª–µ–≤–∞—è –≥–∏–ø–æ—Ç–µ–∑–∞ –≤ —ç—Ç–æ–º —Ç–µ—Å—Ç–µ –∑–∞–∫–ª—é—á–∞–µ—Ç—Å—è –≤ —Ç–æ–º, —á—Ç–æ —Ä–∞—Å–ø—Ä–µ–¥–µ–ª–µ–Ω–∏—è –∑–Ω–∞—á–µ–Ω–∏–π –≤–æ –≤—Å–µ—Ö –≥—Ä—É–ø–ø–∞—Ö —Ä–∞–≤–Ω—ã.

–ù–∞ –æ—Å–Ω–æ–≤–∞–Ω–∏–∏ –ø–æ–ª—É—á–µ–Ω–Ω—ã—Ö —Ä–µ–∑—É–ª—å—Ç–∞—Ç–æ–≤ –º–æ–∂–Ω–æ —Å–¥–µ–ª–∞—Ç—å –≤—ã–≤–æ–¥, —á—Ç–æ —Å—É—â–µ—Å—Ç–≤—É–µ—Ç —Å—Ç–∞—Ç–∏—Å—Ç–∏—á–µ—Å–∫–∏ –∑–Ω–∞—á–∏–º–∞—è —Ä–∞–∑–Ω–∏—Ü–∞ –º–µ–∂–¥—É –≥—Ä—É–ø–ø–∞–º–∏. –ó–Ω–∞—á–µ–Ω–∏–µ p-value –º–µ–Ω—å—à–µ —É—Ä–æ–≤–Ω—è –∑–Ω–∞—á–∏–º–æ—Å—Ç–∏ 0.05, —á—Ç–æ –ø–æ–∑–≤–æ–ª—è–µ—Ç –æ—Ç–≤–µ—Ä–≥–Ω—É—Ç—å –Ω—É–ª–µ–≤—É—é –≥–∏–ø–æ—Ç–µ–∑—É. –≠—Ç–æ –æ–∑–Ω–∞—á–∞–µ—Ç, —á—Ç–æ —Ä–∞—Å–ø—Ä–µ–¥–µ–ª–µ–Ω–∏—è –∑–Ω–∞—á–µ–Ω–∏–π –≤ —Ä–∞–∑–Ω—ã—Ö –≥—Ä—É–ø–ø–∞—Ö —Å—Ç–∞—Ç–∏—Å—Ç–∏—á–µ—Å–∫–∏ –æ—Ç–ª–∏—á–∞—é—Ç—Å—è, –∏ —Å—É—â–µ—Å—Ç–≤—É–µ—Ç –≤–µ—Ä–æ—è—Ç–Ω–æ—Å—Ç—å, —á—Ç–æ —ç—Ç–∏ —Ä–∞–∑–ª–∏—á–∏—è –Ω–µ —Å–ª—É—á–∞–π–Ω—ã.

–ú–µ–∂–¥—É –≥—Ä—É–ø–ø–∞–º–∏ "Coal Mine Terricon" –∏ "Embryo Sand" –Ω–∞–±–ª—é–¥–∞–µ—Ç—Å—è —Å—Ç–∞—Ç–∏—Å—Ç–∏—á–µ—Å–∫–∏ –∑–Ω–∞—á–∏–º–∞—è —Ä–∞–∑–Ω–∏—Ü–∞ (p-value = 0.020921), —Å —É—á–µ—Ç–æ–º –ø–æ–ø—Ä–∞–≤–∫–∏ –Ω–∞ –º–Ω–æ–∂–µ—Å—Ç–≤–µ–Ω–Ω—ã–µ —Å—Ä–∞–≤–Ω–µ–Ω–∏—è (q-value = 0.024140).
–ú–µ–∂–¥—É –≥—Ä—É–ø–ø–∞–º–∏ "Litostrat" –∏ "Embryo Sand" —Ç–∞–∫–∂–µ –Ω–∞–±–ª—é–¥–∞–µ—Ç—Å—è —Å—Ç–∞—Ç–∏—Å—Ç–∏—á–µ—Å–∫–∏ –∑–Ω–∞—á–∏–º–∞—è —Ä–∞–∑–Ω–∏—Ü–∞ (p-value = 0.083265), –Ω–æ –ø–æ—Å–ª–µ –∫–æ—Ä—Ä–µ–∫—Ü–∏–∏ –Ω–∞ –º–Ω–æ–∂–µ—Å—Ç–≤–µ–Ω–Ω—ã–µ —Å—Ä–∞–≤–Ω–µ–Ω–∏—è —ç—Ç–∞ —Ä–∞–∑–Ω–∏—Ü–∞ —Å—Ç–∞–Ω–æ–≤–∏—Ç—Å—è –º–µ–Ω–µ–µ –∑–Ω–∞—á–∏–º–æ–π (q-value = 0.089212).
–ú–µ–∂–¥—É –¥—Ä—É–≥–∏–º–∏ –ø–∞—Ä–∞–º–∏ –≥—Ä—É–ø–ø —Ç–∞–∫–∂–µ –Ω–∞–±–ª—é–¥–∞—é—Ç—Å—è –∑–Ω–∞—á–µ–Ω–∏—è —Å—Ç–∞—Ç–∏—Å—Ç–∏–∫–∏ H, p-value –∏ q-value, –∫–æ—Ç–æ—Ä—ã–µ —É–∫–∞–∑—ã–≤–∞—é—Ç –Ω–∞ —Ä–∞–∑–ª–∏—á–∏—è –º–µ–∂–¥—É –≥—Ä—É–ø–ø–∞–º–∏, –Ω–æ —Ç–æ—á–Ω–æ—Å—Ç—å —ç—Ç–∏—Ö —Ä–∞–∑–ª–∏—á–∏–π –º–æ–∂–µ—Ç –±—ã—Ç—å –Ω–µ—Å–∫–æ–ª—å–∫–æ –Ω–∏–∂–µ –∏–∑-–∑–∞ –∫–æ—Ä—Ä–µ–∫—Ü–∏–∏ –Ω–∞ –º–Ω–æ–∂–µ—Å—Ç–≤–µ–Ω–Ω—ã–µ —Å—Ä–∞–≤–Ω–µ–Ω–∏—è.


```python
# Beta-diversity significance
!qiime diversity beta-group-significance \
  --i-distance-matrix core-metrics-results/bray_curtis_distance_matrix.qza \
  --m-metadata-file map.tsv \
  --m-metadata-column Source\
  --o-visualization core-metrics-results/bray_curtis-body-site-significance.qzv \
  --p-pairwise
```

    [32mSaved Visualization to: core-metrics-results/bray_curtis-body-site-significance.qzv[0m
    [0m

![bray_curtis_body_site_significance1](bray_curtis_body_site_significance1.png)
![bray_curtis_body_site_significance2](bray_curtis_body_site_significance2.png)
![permanova_results](permanova_results.png)

–ê–Ω–∞–ª–∏–∑ PERMANOVA –ø–æ–¥—Ç–≤–µ—Ä–∂–¥–∞–µ—Ç, —á—Ç–æ —Å—É—â–µ—Å—Ç–≤—É–µ—Ç —Å—Ç–∞—Ç–∏—Å—Ç–∏—á–µ—Å–∫–∏ –∑–Ω–∞—á–∏–º–∞—è —Ä–∞–∑–Ω–∏—Ü–∞ –º–µ–∂–¥—É –≥—Ä—É–ø–ø–∞–º–∏ –Ω–∞ –æ—Å–Ω–æ–≤–µ –∞–Ω–∞–ª–∏–∑–∞ –º–Ω–æ–≥–æ–º–µ—Ä–Ω—ã—Ö –¥–∞–Ω–Ω—ã—Ö.
